lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -46,26 +51,89 @@ using namespace llvm;
 using namespace lgc;
 
 namespace llvm {
-
 namespace cl {
 
 // -inreg-esgs-lds-size: Add a dummy "inreg" argument for ES-GS LDS size, this is to keep consistent with PAL's
 // GS on-chip behavior. In the future, if PAL allows hardcoded ES-GS LDS size, this option could be deprecated.
 opt<bool> InRegEsGsLdsSize("inreg-esgs-lds-size", desc("For GS on-chip, add esGsLdsSize in user data"), init(true));
 
 } // namespace cl
-
 } // namespace llvm
 
-namespace lgc {
+namespace {
+
+// =====================================================================================================================
+// The entry-point mutation pass
+class PatchEntryPointMutate : public Patch {
```

Comment:
why do you need remove the name space?

Reply:
There is no need for out-of-line method definitions to be done inside their namespace. Also, if you define a previously-declared function (lgc::createPatchEntryPointMutate here), the LLVM coding standards doc specifically says do not do that inside the namespace, so the build catches any error earlier.